### PR TITLE
Atlas Cloudwatch: Add the FirehoseMetric case class for storing info

### PR DESCRIPTION
**Pull request by manolama**
parsed from the firehose Cloud Watch stream.
Also add an OpenHFT XXHash implementation utility for quickly calculating a unique hash on the Cloud Watch metric for caching.

### CODE REVIEW

File: atlas-cloudwatch/src/main/scala/com/netflix/atlas/util/XXHasher.scala [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+ * limitations under the License.
+ */
+package com.netflix.atlas.util
+
+/**
+  * Simple wrapper and utility functions for hashing with the OpenHFT XXHash implementation.
+  */
+object XXHasher {
+
+  private val hasher = LongHashFunction.xx
```
Comment (brharrington) [inline atlas-cloudwatch/src/main/scala/com/netflix/atlas/util/XXHasher.scala:23]: I think the build failure may be that `LongHashFunction` wasn't imported.

File: atlas-cloudwatch/src/main/scala/com/netflix/atlas/cloudwatch/FirehoseMetric.scala [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+  namespace: String,
+  metricName: String,
+  dimensions: List[Dimension],
+  datapoint: Datapoint
+) {
+
+  def xxHash: Long = {
+    var hash = XXHasher.hash(namespace)
+    hash = XXHasher.updateHash(hash, metricName)
+    dimensions.sortBy(_.name()).foreach { d =>
```
Comment (brharrington) [inline atlas-cloudwatch/src/main/scala/com/netflix/atlas/cloudwatch/FirehoseMetric.scala:36]: How often do you expect this hash will be computed for a given object? It might be better to pre-sort the list when creating the object.

File: atlas-cloudwatch/src/main/scala/com/netflix/atlas/cloudwatch/FirehoseMetric.scala [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+  */
+case class FirehoseMetric(
+  metricStreamName: String,
+  namespace: String,
+  metricName: String,
+  dimensions: List[Dimension],
+  datapoint: Datapoint
+) {
+
+  def xxHash: Long = {
```
Comment (brharrington) [inline atlas-cloudwatch/src/main/scala/com/netflix/atlas/cloudwatch/FirehoseMetric.scala:33]: Why is this needed for this use-case as opposed to the simple built in hash? Maybe add that context as a comment.

File: atlas-cloudwatch/src/test/scala/com/netflix/atlas/util/XXHasherSuite.scala [hunk 4]
```diff
@@ -0,0 +1,67 @@
...
+ * limitations under the License.
+ */
+package com.netflix.atlas.util
+
+import munit.FunSuite
+
+class XXHasherSuite extends FunSuite {
+
+  test("hash byte array") {
+    assertEquals(XXHasher.hash("Hello World".getBytes("UTF-8")), 7148569436472236994L)
```
Comment (brharrington) [inline atlas-cloudwatch/src/test/scala/com/netflix/atlas/util/XXHasherSuite.scala:23]: Is the xx hash implementation guaranteed to be stable? If they tweak values over time to improve characteristics these tests could be flakey for any updates.

File: atlas-cloudwatch/src/main/scala/com/netflix/atlas/util/XXHasher.scala (hunk 1 above)
Comment (manolama) [reply atlas-cloudwatch/src/main/scala/com/netflix/atlas/util/XXHasher.scala:25]: Yup, copy pasta before adding the dep.

File: atlas-cloudwatch/src/test/scala/com/netflix/atlas/util/XXHasherSuite.scala (hunk 4 above)
Comment (manolama) [reply atlas-cloudwatch/src/test/scala/com/netflix/atlas/util/XXHasherSuite.scala:23]: Yes, it will be stable and if something _does_ inadvertently change then we want these UTs to fail so we can roll back or choose a different implementation going forward.

File: atlas-cloudwatch/src/main/scala/com/netflix/atlas/cloudwatch/FirehoseMetric.scala (hunk 3 above)
Comment (manolama) [reply atlas-cloudwatch/src/main/scala/com/netflix/atlas/cloudwatch/FirehoseMetric.scala:33]: Done.

**Review by brharrington — APPROVED** [review]
